src/state/swap/hooks.tsx

```diff
@@ -185,46 +185,57 @@ export function useDerivedSwapInfo(toggledVersion: Version | undefined): {
 
   const bestTrade = isV2TradeBetter === undefined ? undefined : isV2TradeBetter ? v2Trade : v3Trade.trade
 
-  const currencyBalances = {
-    [Field.INPUT]: relevantTokenBalances[0],
-    [Field.OUTPUT]: relevantTokenBalances[1],
-  }
+  const currencyBalances = useMemo(
+    () => ({
+      [Field.INPUT]: relevantTokenBalances[0],
+      [Field.OUTPUT]: relevantTokenBalances[1],
+    }),
+    [relevantTokenBalances]
+  )
 
-  const currencies: { [field in Field]?: Currency | null } = {
```

Comment:
@zzmp Just so i understand here - the issue here is that `currencies` will change its value every render? (if we reference currencies anywhere else) 

But if we only destructure currencies then its fine? (aka no use of currencies itself)

Reply:
exactly this, yes